packages/fiori/src/ViewSettingsDialog.js

```diff
@@ -0,0 +1,375 @@
+import Button from "@ui5/webcomponents/Button.js";
+import GroupHeaderListItem from "@ui5/webcomponents/dist/GroupHeaderListItem.js";
+import List from "@ui5/webcomponents/dist/List.js";
+import StandardListItem from "@ui5/webcomponents/dist/StandardListItem.js";
+import Bar from "./Bar.js";
+
+import {
+	VSD_DIALOG_TITLE_SORT,
+	VSD_SUBMIT_BUTTON,
+	VSD_CANCEL_BUTTON,
+	VSD_RESET_BUTTON,
+	VSD_SORT_ORDER,
+	VSD_SORT_BY,
+	VSD_ORDER_ASCENDING,
+	VSD_ORDER_DESCENDING,
+} from "./generated/i18n/i18n-defaults.js";
+
+// Template
+import ViewSettingsDialogTemplate from "./generated/templates/ViewSettingsDialogTemplate.lit.js";
+
+// Styles
+import viewSettingsDialogCSS from "./generated/themes/ViewSettingsDialog.css.js";
+
+const metadata = {
+	tag: "ui5-view-settings-dialog",
+	managedSlots: true,
+	properties: /** @lends  sap.ui.webcomponents.fiori.ViewSettingsDialog.prototype */ {
+		/**
+		 * Defines the initial sort order.
+		 *
+		 * @type {boolean}
+		 * @defaultvalue false
+		 * @public
+		 */
+		 sortDescending: {
+			type: Boolean,
+		},
+		/**
+		 * Keeps recently focused list in order to focus it on next dialog open.
+		 *
+		 * @type {Object}
+		 * @private
+		 */
+		 _recentlyFocused: {
+			type: Object,
+
+		},
+		/**
+		 * Stores settings of the dialog before the first open.
+		 *
+		 * @type {Object}
+		 * @private
+		 */
+		 _initialSettings: {
+			type: Object,
+		},
+		/**
+		 * Stores settings of the dialog after confirmation.
+		 *
+		 * @type {Object}
+		 * @private
+		 */
+		 _confirmedSettings: {
+			type: Object,
+		},
+		/**
+		 * Stores current settings of the dialog.
+		 *
+		 * @type {Object}
+		 * @private
+		 */
+		 _currentSettings: {
+			type: Object,
+		},
+	},
+	slots: /** @lends  sap.ui.webcomponents.fiori.ViewSettingsDialog.prototype */ {
+		/**
+		 * Defines the <code>sortItems</code> list.
+		 * @type {sap.ui.webcomponents.fiori.ListItem}
+		 * @slot sortItems
+		 * @public
+		 */
+		 "sortItems": {
+			propertyName: "sortItems",
+			type: HTMLElement,
+		},
+	},
+	events: /** @lends  sap.ui.webcomponents.fiori.ViewSettingsDialog.prototype */ {
+
+		/**
+		 * Fired when confirmation button is activated.
+		 *
+		 * @event sap.ui.webcomponents.fiori.ViewSettingsDialog#confirm
+		 * @param {String} sortOrder The current sort order selected.
+		 * @param {String} sortBy The current sort by selected.
+		 * @public
+		 */
+		"confirm": {
+			detail: {
+				sortOrder: { type: String },
+				sortBy: { type: String },
+			},
+		},
+		/**
+		 * Fired when Cancel button is activated.
+		 *
+		 * @event sap.ui.webcomponents.fiori.ViewSettingsDialog#cancel
+		 * @param {Object} settings The current settings.
+		 * @public
+		 */
+		"cancel": {
+			detail: {
+				confirmedSortOrder: { type: String },
+				confirmedSortBy: { type: String },
+			},
+		},
+	},
+};
+
+/**
+ * @class
+ * <h3 class="comment-api-title">Overview</h3>
+ * The <code>ui5-view-settings-dialog</code> component consisting of several lists.
```

Comment:
Let's start with a description of the component like:
* The <code>ui5-view-settings-dialog</code> component helps the user to sort, filter, or group data within a list or a table.

Reply:
Done